### PR TITLE
473 475 client managment

### PR DESCRIPTION
**Pull request by wundersys**
swagger: https://localhost:62189/identity/swagger/index.html
access token url: https://localhost:62189/identity/connect/token

### CODE REVIEW

**Comment by fw2568** [comment]
Hello Tolga,

please rebase to latest commits on master to resolve conflicts.

Thanks,
Frank

File: src/Haipa.Modules.Identity/IdentityModule.cs [hunk 1]
```diff
@@ -52,29 +117,69 @@ protected override void ConfigureServices(IServiceProvider serviceProvider, ISer
...
+                options.AddPolicy("CorsPolicy",
+                    builder => builder.AllowAnyOrigin()
+                        .AllowAnyMethod()
+                        .AllowAnyHeader()
+                        .AllowCredentials());
             });
-            services.AddAuthorization(options =>
+            services.Configure<MvcOptions>(options =>
             {
-                options.AddPolicy("identity:apps:write:all", policy => policy.Requirements.Add(new HasScopeRequirement("identity:apps:write:all", "http://localhost:62189/identity")));
```
Comment (dbosoft-fw) [inline src/Haipa.Modules.Identity/IdentityModule.cs]: Question: why you had to add this?

File: src/Haipa.Modules.Identity/IdentityModule.cs [hunk 2]
```diff
@@ -52,29 +117,69 @@ protected override void ConfigureServices(IServiceProvider serviceProvider, ISer
             //        options.RequireHttpsMetadata = false;
             //    });
 
-            services.AddAuthorization(options =>
```
Comment (dbosoft-fw) [inline src/Haipa.Modules.Identity/IdentityModule.cs]: With these changes there is currently no auth in identity service. But authentication will also be required for management of clients.

File: src/Haipa.IdentityDb/IDbContextConfigurer.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Haipa.IdentityDb
+{ 
+    public interface IDbContextConfigurer<TContext> where TContext : DbContext
+    {
+        void Configure(DbContextOptionsBuilder options);
+    }
+    public interface ITenantProvider
```
Comment (dbosoft-fw) [inline src/Haipa.IdentityDb/IDbContextConfigurer.cs]: please remove this interface

File: src/Haipa.IdentityDb/Stores/ClientStore.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Haipa.IdentityDb.Stores
+{
+    public class ClientStore : IClientStore
```
Comment (dbosoft-fw) [inline src/Haipa.IdentityDb/Stores/ClientStore.cs]: Is that really necessary? Does the Identity Server not already implement an EFCore based store?

File: src/Haipa.Modules.Identity/IdentityModule.cs (hunk 1 above)
Comment (wundersys) [reply src/Haipa.Modules.Identity/IdentityModule.cs]: *see my comment above please

File: src/Haipa.Modules.Identity/IdentityModule.cs (hunk 2 above)
Comment (wundersys) [reply src/Haipa.Modules.Identity/IdentityModule.cs]: please let us speak the authorization in detail

File: src/Haipa.IdentityDb/Stores/ClientStore.cs (hunk 4 above)
Comment (wundersys) [reply src/Haipa.IdentityDb/Stores/ClientStore.cs]: the clientstore was modified with a custom one (and models)

**Comment by fw2568** [comment]
Please rebase to PR #14 and update tests / code if tests are unsucessful

**Pull request merged**